packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation.js

```diff
@@ -105,6 +113,19 @@ export const usePaymentInformation = props => {
         [onSave]
     );
 
+    const clearPaymentDetails = useCallback(() => {
+        client.writeQuery({
+            query: getPaymentNonceQuery,
+            data: {
+                cart: {
+                    __typename: 'Cart',
+                    id: cartId,
+                    paymentNonce: null
```

Comment:
I don't think that `usePaymentInformation` should know about the nonce as it is specific to braintree/credit card.

That said, after working on payment information I understand the component heirarchy a bit more. `CreditCard` is not mounted if you are in a "done editing" view which means that there's no way to pass the error to that component and have it wipe CC specific data like nonce or billing address.

Reply:
Exactly. Also, it is not a rule but it can be an understanding that any payment method which has to save something related to payment on the cart in the cache, should place it in the key called `paymentNonce`. That way, we can handle clearing off of the cache for any payment method in the payment information component.

Reply:
>  any payment method which has to save something related to payment on the cart in the cache, should place it in the key called `paymentNonce`.

I like the idea here but I don't think `paymentNonce` is a good name for it 😄Probably worth a small refactor here to rename this property. Perhaps `selectedPaymentDetails`?

Reply:
@sirugh should we still change the prop? If so, I would like to handle it as part of a diff PR. I would not like to expand the scope of this PR. What do you think?

Reply:
I think `paymentNonce` should be fine as a name. I'd also be concerned about tearing that name out of every place we have it currently.